src/AlignLoads.cpp

```diff
@@ -0,0 +1,210 @@
+#include "AlignLoads.h"
+#include "IRMutator.h"
+#include "IROperator.h"
+#include "Scope.h"
+#include "Bounds.h"
+#include "ModulusRemainder.h"
+#include "Simplify.h"
+namespace Halide {
+namespace Internal {
+using std::vector;
+
+class AlignLoads : public IRMutator {
+public:
+    AlignLoads(Target t, int vs) : target(t), required_alignment(vs) {}
+private:
+    Target target;
+    // The desired alignment of a vector load.
+    int required_alignment;
+    // Alignment info for variables in scope.
+    Scope<ModulusRemainder> alignment_info;
+    using IRMutator::visit;
+    ModulusRemainder get_alignment_info(Expr e) {
+        return modulus_remainder(e, alignment_info);
+    }
+    int natural_vector_lanes(Type t) {
```

Comment:
I think this should now be called something else. Maybe just get rid of it? It's only used once, and I can't think of a reasonable name for it.